src/wallet/wallet.h

```diff
@@ -250,11 +250,13 @@ class CAddressBookIterator
     explicit CAddressBookIterator(std::map<CTxDestination, AddressBook::CAddressBookData>& _map) : map(_map), it(_map.begin()), itEnd(_map.end()) {}
     CTxDestination GetKey() { return it->first; }
     AddressBook::CAddressBookData GetValue() { return it->second; }
-    bool HasNext() { return it != itEnd; }
+
+    bool IsValid() { return it != itEnd && it != map.end(); }
```

Comment:
(e0ffc983ce0af18cdd1584176a7b7b5b581d8d5d)
This change confuses me.
If we set the filter and advance with `Next()`, how can we get past `itEnd` and reach `map.end()` (if they are different)?

Reply:
With the current code, we can't yeah. I added it previously with another subfiltering logic in mind (when the `SetFilter` wasn't a `map.find` only). can remove it 👍 .